codec/src/main/java/io/netty/handler/codec/DelimitedLengthFieldBasedFrameDecoder.java

```diff
@@ -0,0 +1,179 @@
+    private long frameLength;
+
+    public DelimitedLengthFieldBasedFrameDecoder(
+            int maxFrameLength,
+            int lengthAdjustment,
+            boolean failFast,
+            ByteBuf delimiter,
+            Charset lengthFieldCharset,
+            boolean trimLengthString
+    ) {
+        super(maxFrameLength, failFast);
+        if (delimiter == null) {
+            throw new NullPointerException("delimiter");
+        }
+        if (!delimiter.isReadable()) {
+            throw new IllegalArgumentException("empty delimiter");
+        }
+
+        this.delimiter = delimiter.slice(delimiter.readerIndex(), delimiter.readableBytes());
+        this.lengthFieldCharset = lengthFieldCharset;
+
+        this.lengthAdjustment = lengthAdjustment;
+        this.trimLengthString = trimLengthString;
+    }
+
+    /**
+     * Returns the number of bytes between the readerIndex of the haystack and
+     * the first needle found in the haystack.  -1 is returned if no needle is
+     * found in the haystack.
+     * <p>
+     * For now, this delegates to the implementation within
+     * {@link DelimiterBasedFrameDecoder#indexOf(ByteBuf, ByteBuf)}
+     *
+     * @param haystack the buffer to search for an occurence of the needle
+     * @param needle the buffer to find in the haystack
+     *
+     * @return the first byte index of the needle found in the haystack, or -1 if not found
+     */
+    protected int indexOf(ByteBuf haystack, ByteBuf needle) {
+        return DelimiterBasedFrameDecoder.indexOf(haystack, needle);
+    }
+
+    /**
+     * Create a frame out of the {@link ByteBuf} and return it.
+     *
+     * @param ctx the {@link ChannelHandlerContext} which this {@link ByteToMessageDecoder} belongs to
+     * @param in the {@link ByteBuf} from which to read data
+     *
+     * @return the {@link ByteBuf} which represents the frame or {@code null} if no frame could be created.
+     */
+    @Override
+    protected Object decode(ChannelHandlerContext ctx, ByteBuf in) throws Exception {
+        if (discardingTooLongFrame) {
+            long bytesToDiscard = this.bytesToDiscard;
+            int localBytesToDiscard = (int) Math.min(bytesToDiscard, in.readableBytes());
+            in.skipBytes(localBytesToDiscard);
+            bytesToDiscard -= localBytesToDiscard;
+            this.bytesToDiscard = bytesToDiscard;
+
+            failIfNecessary(false);
+
+            // if exception was not thrown, reader has been advanced by readable bytes already, so return null to
+            // continue processing too long frame
+            // TODO: check whether also needed in LengthFieldBasedFrameDecoder; I believe it is
+            return null;
+        }
+
+        if (consumingLength) {
+            int delimIndex = indexOf(in, delimiter);
+            if (delimIndex < 0) {
+                return null;
+            }
+
+            final String lengthStr = in.toString(in.readerIndex(), delimIndex, lengthFieldCharset);
```

Comment:
delimIndex is actually already a length measure.  It comes from indexOf, which effectively finds the index of the delimiter, then subtracts the readerIndex.  I am adding a new unit test which asserts this (and fails when subtracting the readerIndex).

I'm not sure I follow about the DDoS.  Can you please elaborate a bit there?